### PR TITLE
[SAK-29342] Allow a user to unjoin a joinable site

### PR DESCRIPTION
**Pull request by adrianfish**
From the site info tool.

### CODE REVIEW

File: site-manage/site-manage-tool/tool/src/bundle/sitesetupgeneric.properties [hunk 1]
```diff
@@ -627,6 +627,9 @@ sitinfimp.youcan.user1 = You can choose to import user(s) only from other
 sitinfimp.youcan.user2 = sites that you own. You can combine users from more than one site.
 sitinfimp.material.hidden.warn = Content imported from the Resources tool will be marked as hidden (indicated by grey text), meaning it will not be visible to site participants who do not have permission to access hidden content. <br/> By default, 'guest' and 'student' roles cannot view items marked as hidden, although a site administrator can modify this default setting.
 sitinfimp.material.visible.warn = Content imported from the Resources tool will be visible by default, meaning it will be accessible to site participants like 'guest' and 'student' roles, although a site administrator can modify this setting.
+sitinfimp.youhave = You are no longer a participant in site
```
Comment (steveswinsburg) [inline site-manage/site-manage-tool/tool/src/bundle/sitesetupgeneric.properties]: Do we want fullstops here?

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java [hunk 2]
```diff
@@ -15151,6 +15156,54 @@ private Locale getLocaleFromString(String localeString)
...
+			.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
+		final ParameterParser params = data.getParameters();
+
+		final String id = params.get("itemReference");
+		String siteTitle = null;
+		
+		if (id != null)	{
+			try	{
+				siteTitle = SiteService.getSite(id).getTitle();
+                System.out.println("siteTitle:" + siteTitle);
```
Comment (steveswinsburg) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: println, perhaps logging :)

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java [hunk 3]
```diff
@@ -15151,6 +15156,54 @@ private Locale getLocaleFromString(String localeString)
...
+		// refresh the whole page
+		scheduleTopRefresh();
+		
+		/*
+		 * It would have been nice to redirect to the portal 
+		 * if the user no longer has access to view the site;
+		 * but it seems that it is not possible in velocity
+		 * to redirect if action is set.
+		 * 
+		*/
```
Comment (steveswinsburg) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: Based on this comment, what happens when they are unjoined? What do they see?

**Comment by steveswinsburg** [comment]
Looks good, just a few comments inline. Cheers.

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java (hunk 3 above)
Comment (adrianfish) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: They get redirected to the same site and get the join dialog - I'll just strip that comment out. It dates back to 2.5.

**Pull request merged**